### ISSUE TITLE
Docker desktop volume mount mapping with WSL not working

### ISSUE THREAD

**Issue by Depechie** [docker/for-win#13014]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 44D309F6-5487-48F1-A069-9B43399E6E9F/20221023112433

### Actual behavior

When I use a docker compose file and try to bind mount files into the container with the following syntax:

```
    volumes:
      - /var/lib/docker/volumes/otel/blackbox.yml:/etc/blackbox/blackbox.yml
```

The Ubuntu running with WSL2 context will auto create the folder `/var/lib/docker/volumes/otel`. But this mapping is not available in the windows host.

### Expected behavior

On all my other windows devices with the same setup, the `/var/lib/docker/volumes` is automapped to `\\wsl$\docker-desktop-data\data\docker\volumes` instead of docker creating the actual `/var/lib/docker/volumes` folder.

I already tried reinstalling WSL and Docker Desktop but without any effects.
Is there something missing with my new install?

### Information
  - Windows Version: Windows 11 Enterprise 21H2
  - Docker Desktop Version: v4.12.0
  - WSL2 or Hyper-V backend?  WSL2

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`

```
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0017: can a VM be started?
[PASS] DD0024: is WSL installed?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0007: is the backend responding?
[FAIL] DD0014: are the backend processes running? 1 error occurred:
        * vpnkit.exe is not running


[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[PASS] DD0012: is the VM networking working?
[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 1 issue:

1 : The test: are the backend processes running?
    Failed with: 1 error occurred:
        * vpnkit.exe is not running



Not all of the backend processes are running.
```

**Comment by david-engelmann** [comment]
@Depechie What are the results of `docker inspect blah`?

**Comment by Depechie** [comment]
hey @david-engelmann for reference I will provide both versions
This is the one of the blackbox docker container that is not using the docker-desktop-data:

```
[
    {
        "Id": "69332459f381259430cab09b805065b65fc41e1a7471b8788dd20db2dfc5b32d",
        "Created": "2022-10-20T20:35:20.2395341Z",
        "Path": "/bin/blackbox_exporter",
        "Args": [
            "--config.file=/etc/blackbox/blackbox.yml"
        ],
        "State": {
            "Status": "restarting",
            "Running": true,
            "Paused": false,
            "Restarting": true,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2022-10-20T20:35:28.5047267Z",
            "FinishedAt": "2022-10-20T20:35:28.5410104Z"
        },
        "Image": "sha256:4a42686b55103f19d24918dd77fd38e29e4fe200cf861e0c32fb2f4701791d2c",
        "ResolvConfPath": "/var/lib/docker/containers/69332459f381259430cab09b805065b65fc41e1a7471b8788dd20db2dfc5b32d/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/69332459f381259430cab09b805065b65fc41e1a7471b8788dd20db2dfc5b32d/hostname",
        "HostsPath": "/var/lib/docker/containers/69332459f381259430cab09b805065b65fc41e1a7471b8788dd20db2dfc5b32d/hosts",
        "LogPath": "/var/lib/docker/containers/69332459f381259430cab09b805065b65fc41e1a7471b8788dd20db2dfc5b32d/69332459f381259430cab09b805065b65fc41e1a7471b8788dd20db2dfc5b32d-json.log",
        "Name": "/blackbox",
        "RestartCount": 6,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/var/lib/docker/volumes/blackbox.yml:/etc/blackbox/blackbox.yml:rw"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "stacktest_default",
            "PortBindings": {
                "9115/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "9115"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "unless-stopped",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": [],
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/3624f333f841b29681031c6eb4f48c5f1686df11c3ce0071d18613ecf6813e38-init/diff:/var/lib/docker/overlay2/07ddd8397357a3bab10b5d52d4de980ffa9df1c274011422cf9ed5865b53977b/diff:/var/lib/docker/overlay2/10bd3445d783d73472077dd35f716f38390df60288702b1e50952e1a643b9cfd/diff:/var/lib/docker/overlay2/5455d18bc81ba6f5bc6868906b93a94ae733b38cbe8ac71b6a0f27087fef7b05/diff:/var/lib/docker/overlay2/2dd199dd661fcf121754bf8e5cd950f8207fdc24f269d83348b670d2edee27f5/diff",
                "MergedDir": "/var/lib/docker/overlay2/3624f333f841b29681031c6eb4f48c5f1686df11c3ce0071d18613ecf6813e38/merged",
                "UpperDir": "/var/lib/docker/overlay2/3624f333f841b29681031c6eb4f48c5f1686df11c3ce0071d18613ecf6813e38/diff",
                "WorkDir": "/var/lib/docker/overlay2/3624f333f841b29681031c6eb4f48c5f1686df11c3ce0071d18613ecf6813e38/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/var/lib/docker/volumes/blackbox.yml",
                "Destination": "/etc/blackbox/blackbox.yml",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "69332459f381",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": true,
            "AttachStderr": true,
            "ExposedPorts": {
                "9115/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "--config.file=/etc/blackbox/blackbox.yml"
            ],
            "Image": "prom/blackbox-exporter:v0.22.0",
            "Volumes": {
                "/etc/blackbox/blackbox.yml": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "/bin/blackbox_exporter"
            ],
            "OnBuild": null,
            "Labels": {
                "com.docker.compose.config-hash": "159a62e10a95e2abfb138afa67dc7adecd621d2358280637628930b4896a4364",
                "com.docker.compose.container-number": "1",
                "com.docker.compose.depends_on": "",
                "com.docker.compose.oneoff": "False",
                "com.docker.compose.project": "stacktest",
                "com.docker.compose.project.config_files": "/data/compose/1/docker-compose.yml",
                "com.docker.compose.project.working_dir": "/data/compose/1",
                "com.docker.compose.service": "blackbox",
                "com.docker.compose.version": "2.5.1",
                "desktop.docker.io/wsl-distro": "Ubuntu",
                "maintainer": "The Prometheus Authors \u003cprometheus-developers@googlegroups.com\u003e"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "273cb7f47759875b15ccc17924aeca91fd5de9c518601043f8a7828162ea4ecb",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/273cb7f47759",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "stacktest_default": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "blackbox",
                        "blackbox",
                        "69332459f381"
                    ],
                    "NetworkID": "e4022ad5a261c29bb7c4e1ba2a920bc9485a147aaa9d0ab95327f506b4b09081",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Comment by david-engelmann** [comment]
It's possible it's your source path that is causing the issue
Working:
```      
{
"Type": "bind",
"Source": "/var/lib/docker/volumes/otel/blackbox.yml",
"Destination": "/etc/blackbox/blackbox.yml",
"Mode": "rw",
"RW": true,
"Propagation": "rslave"
},
```
Not Working:
```
{
"Type": "bind",
"Source": "/var/lib/docker/volumes/blackbox.yml",
"Destination": "/etc/blackbox/blackbox.yml",
"Mode": "rw",
"RW": true,
"Propagation": "rprivate"
}
```
Notice the differences between the Propagation.
Other differences are around the networking of the two configurations

as far as the differences between the ResolvConfPath, HostnamePath, HostsPath, LogPath, there doesn't seem to be an issue. If you are still having issues, make sure docker-desktop-data is running with the `wsl -l -v` command from the command prompt

**Comment by Depechie** [comment]
@david-engelmann the paths differ because I have been trying out different compose files to see if it works or not.
But if I use a compose file with following syntax:

```
services:
  blackbox:
    image: prom/blackbox-exporter:v0.22.0
    container_name: blackbox
    restart: unless-stopped
    ports:
      - 9115:9115
    volumes:
      - /var/lib/docker/volumes/blackbox.yml:/etc/blackbox/blackbox.yml
    command: --config.file=/etc/blackbox/blackbox.yml
```
I will get a blackbox.yml directory on the windows pc that is not working as I suspect, although I have the correct file in the volumes directory I always put it on working pc's ( `\wsl$\docker-desktop-data\data\docker\volumes` ).

How can I check the correct startup for the docker-desktop-data? I just install docker desktop on windows and have WSL configured in the UI.

![image](https://user-images.githubusercontent.com/351693/199543965-ee8762fe-cea4-4ee1-8909-d9e4986f18b8.png)

![image](https://user-images.githubusercontent.com/351693/199544904-cf0aeb79-c061-4d76-a0fd-2187695e1def.png)

**Comment by david-engelmann** [comment]
@Depechie You can check the status of docker-desktop-data by running the `wsl -l -v` command from a command prompt

**Comment by Depechie** [comment]
@david-engelmann 
![image](https://user-images.githubusercontent.com/351693/200197113-1288f684-2cc8-46ea-a7e0-888c50c2c9ff.png)

**Comment by david-engelmann** [comment]
@Depechie That looks correct! And it sounds like you are declaring the mount in an identical way.  The mount or working directory isn't declared anywhere else.

**Comment by Depechie** [comment]
hey @david-engelmann not that I'm aware of. It is only 1 docker-compose file with only containing that mount.
I registered this issue, because I have indeed no clue what is different for this machine than the 2 other dev machines I'm using where things work as I would suspect them to work. ( that linux mounts to /var/lib/docker/volumes would be 'auto' mapped to the win host at \\wsl$\docker-desktop-data\data\docker\volumes )

And it is halting me in running a workshop script, because that script tries to manage the needed files on the windows host wsl directory.

But I know this problem can be related to other things of course... maybe something with windows itself, or WSL or maybe the Ubuntu install?

Only thing I can mention is that this is a fresh installed pc, everything new installed.
